src/OSPSuite.Infrastructure.Import/Services/IImportFromExcelTask.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using OSPSuite.Infrastructure.Import.Core.DataSourceFileReaders;
+using OSPSuite.Infrastructure.Import.Extensions;
+using OSPSuite.Utility.Exceptions;
+
+namespace OSPSuite.Infrastructure.Import.Services
+{
+   public interface IImportFromExcelTask
+   {
+      /// <summary>
+      /// Retrieves all the appropriate sheet names from an excel file
+      /// </summary>
+      /// <param name="fileName">The filename and path of the excel file</param>
+      /// <param name="excludeEmptySheets">Excludes empty sheets from the list if set to true</param>
+      /// <returns>The list of sheet names</returns>
+      IReadOnlyList<string> RetrieveExcelSheets(string fileName, bool excludeEmptySheets = false);
+
+      /// <summary>
+      /// Gets the DataTable for the filepath, and sheet indicated
+      /// </summary>
+      /// <param name="filePath">The path to the excel file</param>
+      /// <param name="sheetName">the sheet in the file to convert</param>
+      /// <param name="firstRowAsCaption">Whether or not the first row in the sheet should be interpreted as a caption</param>
+      /// <returns>Data table corresponding to the file and sheet</returns>
+      DataTable GetDataTables(string filePath, string sheetName, bool firstRowAsCaption);
+
+      /// <summary>
+      /// Gets all the DataTables from the Excel workbook at the specified filePath
+      /// </summary>
+      /// <param name="filePath">The path to the excel file</param>
+      /// <param name="firstRowAsCaption">Whether or not the first row in the sheet should be interpreted as a caption</param>
+      /// <returns>The data tables corresponding to the file</returns>
+      IReadOnlyList<DataTable> GetAllDataTables(string filePath, bool firstRowAsCaption);
+   }
+
+   public class ImportFromExcelTask : IImportFromExcelTask
+   {
+      public IReadOnlyList<string> RetrieveExcelSheets(string fileName, bool excludeEmptySheets = false)
```

Comment:
why is default false? Does not true make more sense?

Reply:
I have kept this exactly the way it was in MoBi. My understanding is that if you just want the list of names of all the sheets, it is the default that you get even the empty ones, since they exist as sheets.